drivers/InterruptIn.cpp

```diff
@@ -19,16 +19,33 @@
 
 namespace mbed {
 
+// Note: This single-parameter constructor exists to maintain binary
+//       compatibility for the Wi-Fi drivers.
```

Comment:
This would be fine in the commit message but I would not keep this in the source here.

It is not just for wifi driver, it was written as an example that the previous change broke binary compability and we should keep it as far as we can, and here we could do it.